metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py

```diff
@@ -182,10 +188,13 @@ def authenticator_type_is_valid(cls, v, values, field):
             )
         if v == "KEY_PAIR_AUTHENTICATOR":
             # If we are using key pair auth, we need the private key path and password to be set
-            if values.get("private_key_path") is None:
+            if (
+                values.get("private_key") is None
+                and values.get("private_key_path") is None
+            ):
                 raise ValueError(
-                    f"'private_key_path' was none "
-                    f"but should be set when using {v} authentication"
+                    f"Both `private_key` and `private_key_path` are none. "
```

Comment:
nit: i see that we're throwing the error below, but ideally we should just be doing the error checking in one place

Reply:
Yes. This is the main place I intend to keep error checking at. The assertion in below method was added primarily for linter.

Reply:
To remove the assert, we can add `if.. elif..` , however assert seemed cleaner to me.